src/generators/web/utils/processing.mjs

```diff
@@ -1,81 +1,108 @@
-  const { value: code } = toJs(entry, { handlers: jsx });
-
-  // `buildServerProgram` takes the JSX-derived code and prepares it for server execution.
-  // `executeServerCode` then runs this code in a Node.js environment to produce
-  // the initial HTML content (dehydrated state) that will be sent to the client.
-  const serverCode = buildServerProgram(code);
-  const dehydrated = await executeServerCode(serverCode, requireFn);
-
-  // `buildClientProgram` prepares the JSX-derived code for client-side execution.
-  // `bundleCode` then bundles this client-side code, resolving imports and
-  // potentially generating associated CSS. This bundle will hydrate the SSR content.
-  const clientCode = buildClientProgram(code);
-  const clientBundle = await bundleCode(clientCode);
-
-  const title = `${entry.data.heading.data.name} | Node.js v${version} Documentation`;
-
-  // Replace template placeholders with actual content
-  const renderedHtml = template
-    .replace('{{title}}', title)
-    .replace('{{dehydrated}}', dehydrated ?? '')
-    .replace('{{clientBundleJs}}', () => clientBundle.js);
-
-  // The input to `minify` must be a Buffer.
-  const finalHTMLBuffer = HTMLMinifier.minify(Buffer.from(renderedHtml), {});
-
-  // Return the generated HTML and any CSS produced by the client bundle.
+  const serverCodeMap = new Map();
+  const clientCodeMap = new Map();
+
+  // Convert JSX AST to JavaScript for both server and client
+  for (const entry of entries) {
+    const fileName = `${entry.data.api}.jsx`;
+
+    // Convert AST to JavaScript string with JSX syntax
+    const { value: code } = toJs(entry, { handlers: jsx });
+
+    // Prepare code for server-side execution (wrapped for SSR)
+    serverCodeMap.set(fileName, buildServerProgram(code));
+
+    // Prepare code for client-side execution (wrapped for hydration)
+    clientCodeMap.set(fileName, buildClientProgram(code));
+  }
+
+  // Execute all server code at once to get dehydrated HTML
+  const serverBundle = await executeServerCode(serverCodeMap, requireFn);
+
+  // Bundle all client code at once (with code splitting for shared chunks)
+  const clientBundle = await bundleCode(clientCodeMap);
+
+  // Process each entry to create final HTML
+  const results = entries.map(({ data }) => {
+    const fileName = `${data.api}.js`;
+
+    const title = `${data.heading.data.name} | Node.js v${version} Documentation`;
+
+    // Replace template placeholders with actual content
+    const renderedHtml = template
+      .replace('{{title}}', title)
+      .replace('{{dehydrated}}', serverBundle.get(fileName) ?? '')
+      .replace('{{importMap}}', clientBundle.importMapHtml)
+      .replace('{{entrypoint}}', `./${fileName}?${randomUUID()}`);
+
+    // Minify HTML (input must be a Buffer)
+    const finalHTMLBuffer = HTMLMinifier.minify(Buffer.from(renderedHtml), {});
+
+    return { html: finalHTMLBuffer, api: data.api };
+  });
```

Comment:
```suggestion
  const results = entries.map(({ data }) => {
    // Replace template placeholders with actual content
    const renderedHtml = template
      .replace('{{title}}',  `${data.heading.data.name} | Node.js v${version} Documentation`)
      .replace('{{dehydrated}}', serverBundle.get(fileName) ?? '')
      .replace('{{importMap}}', clientBundle.importMapHtml)
      .replace('{{entrypoint}}', `./${data.api}.js${randomUUID()}`);

    // Minify HTML (input must be a Buffer)
    const finalHTMLBuffer = HTMLMinifier.minify(Buffer.from(renderedHtml), {});

    return { html: finalHTMLBuffer, api: data.api };
  });
```

Like I said in my other suggestion, it doesn't make sense to partially inline and partially not. I think we should just inline the entire thing.